cpp/src/unaryop.cpp

```diff
@@ -45,19 +57,56 @@ class CastTask : public Task<CastTask, OpCode::Cast> {
   }
 };
 
+cudf::unary_operator arrow_to_cudf_unary_op(std::string op)
+{
+  // Arrow unary operators taken from the below list,
+  // where an equivalent cudf unary operator exists.
+  // https://arrow.apache.org/docs/cpp/compute.html#element-wise-scalar-functions
+  // https://docs.rapids.ai/api/libcudf/stable/group__transformation__unaryops
+  std::unordered_map<std::string, cudf::unary_operator> arrow_to_cudf_ops = {
+    {"sin", cudf::unary_operator::SIN},       {"cos", cudf::unary_operator::COS},
+    {"tan", cudf::unary_operator::TAN},       {"asin", cudf::unary_operator::ARCSIN},
+    {"acos", cudf::unary_operator::ARCCOS},   {"atan", cudf::unary_operator::ARCTAN},
+    {"sinh", cudf::unary_operator::SINH},     {"cosh", cudf::unary_operator::COSH},
+    {"tanh", cudf::unary_operator::TANH},     {"asinh", cudf::unary_operator::ARCSINH},
+    {"acosh", cudf::unary_operator::ARCCOSH}, {"atanh", cudf::unary_operator::ARCTANH},
+    {"exp", cudf::unary_operator::EXP},       {"ln", cudf::unary_operator::LOG},
+    {"sqrt", cudf::unary_operator::SQRT},     {"ceil", cudf::unary_operator::CEIL},
+    {"floor", cudf::unary_operator::FLOOR},   {"abs", cudf::unary_operator::ABS},
+    {"round", cudf::unary_operator::RINT},    {"bit_wise_not", cudf::unary_operator::BIT_INVERT},
+    {"invert", cudf::unary_operator::NOT},    {"negate", cudf::unary_operator::NEGATE}};
+
+  if (arrow_to_cudf_ops.find(op) != arrow_to_cudf_ops.end()) { return arrow_to_cudf_ops[op]; }
+  throw std::invalid_argument("Could not find cudf binary operator matching: " + op);
+  return cudf::unary_operator::ABS;
```

Comment:
```suggestion
```

Reply:
The compiler will complain if there is no return type and there is no "null" option given by cudf.

Reply:
Ah OK, nvm then. Somehow thought it would be OK behind a throw.